fbw-a380x/src/systems/instruments/src/FCU/Managers/SpeedManager.ts

```diff
@@ -1,17 +1,30 @@
-// Copyright (c) 2023-2024 FlyByWire Simulations
+// Copyright (c) 2023-2025 FlyByWire Simulations
 // SPDX-License-Identifier: GPL-3.0
 
-import { EventBus, Instrument } from '@microsoft/msfs-sdk';
+import { ConsumerSubject, EventBus, Instrument } from '@microsoft/msfs-sdk';
 import { TemporaryHax } from './TemporaryHax';
+import { FGVars } from 'instruments/src/MsfsAvionicsCommon/providers/FGDataPublisher';
+import { VerticalMode } from '@shared/autopilot';
+import { FmgcFlightPhase } from '@shared/flightphase';
 
 // FIXME port to MSFS avionics framework style
 export class SpeedManager extends TemporaryHax implements Instrument {
-  private readonly backToIdleTimeout = 10000;
+  private readonly backToIdleTimeout = 45000;
   private readonly MIN_SPEED = 100;
   private readonly MAX_SPEED = 399;
   private readonly MIN_MACH = 0.1;
   private readonly MAX_MACH = 0.99;
 
+  private readonly sub = this.bus.getSubscriber<FGVars>();
+
+  private readonly managedSpeed = ConsumerSubject.create(this.sub.on('fg.speeds.managed'), 0);
+
+  private readonly validManagedSpeed = this.managedSpeed.map((v) => v >= 90);
+
+  private readonly managedSpeedTarget = ConsumerSubject.create(this.sub.on('fg.speeds.managedTarget'), 0);
+
+  private readonly validManagedSpeedTarget = this.managedSpeedTarget.map((v) => v > 90);
```

Comment:
Do we really need all these subscribables that only subscribe to each other and calculate millions of results that are never used? We could just directly compute from simvars only when actually needed in getInitialSelectedSpeedValue, much like before, which is as few as zero times per flight.